public/app/core/components/SplitPaneWrapper/SplitPaneWrapper.tsx

```diff
@@ -33,94 +26,48 @@ export class SplitPaneWrapper extends PureComponent<Props> {
   }
 
   updateSplitPaneSize = () => {
-    if (this.rafToken.current !== undefined) {
-      window.cancelAnimationFrame(this.rafToken.current!);
+    if (this.refToken.current !== undefined) {
+      window.cancelAnimationFrame(this.refToken.current!);
     }
-    this.rafToken.current = window.requestAnimationFrame(() => {
+    this.refToken.current = window.requestAnimationFrame(() => {
       this.forceUpdate();
     });
   };
 
-  onDragFinished = (pane: Pane, size?: number) => {
+  onDragFinished = (size?: number) => {
     document.body.style.cursor = 'auto';
 
-    // When the drag handle is just clicked size is undefined
-    if (!size) {
-      return;
-    }
-
-    const { updateUiState } = this.props;
-    if (pane === Pane.Top) {
-      updateUiState({
-        topPaneSize: size / window.innerHeight,
-      });
-    } else {
-      updateUiState({
-        rightPaneSize: size / window.innerWidth,
-      });
+    if (this.props.onDragFinished && size !== undefined) {
+      this.props.onDragFinished(size);
     }
   };
 
   onDragStarted = () => {
-    document.body.style.cursor = 'row-resize';
+    document.body.style.cursor = this.props.splitOrientation === 'horizontal' ? 'row-resize' : 'col-resize';
   };
 
-  renderHorizontalSplit() {
-    const { leftPaneComponents, uiState } = this.props;
-    const styles = getStyles(config.theme2);
-    const topPaneSize = uiState.topPaneSize >= 1 ? uiState.topPaneSize : uiState.topPaneSize * window.innerHeight;
-
-    /*
-      Guesstimate the height of the browser window minus
-      panel toolbar and editor toolbar (~120px). This is to prevent resizing
-      the preview window beyond the browser window.
-     */
-
-    if (Array.isArray(leftPaneComponents)) {
-      return (
-        <SplitPane
-          split="horizontal"
-          maxSize={-200}
-          primary="first"
-          size={topPaneSize}
-          pane2Style={{ minHeight: 0 }}
```

Comment:
The refactor does not add this in, because I could not verify it actually works. I believe it is meant such that the non primary pane, in this case the bottom pane, could be resized to 0, but that is not the case. The top pane can be resized to 0. Let me know if I'm missing something here.

Reply:
no this is relates to flexbox and flex-grow , this used be required to get thing working (with scrollable panes), but not sure it's required anymore (was a quirk with css and browsers)

Reply:
Ah, it is actually still needed. The e2e test caught it, but the transform tab doesnt scroll without it.